core/src/main/scala/com/gu/tableversions/core/TableVersions.scala

```diff
@@ -0,0 +1,53 @@
+package com.gu.tableversions.core
+
+import java.time.Instant
+
+import com.gu.tableversions.core.TableVersions.CommitResult
+
+/**
+  * This defines the interface for querying and updating table version information tracked by the system.
+  */
+trait TableVersions[F[_]] {
+
+  /**
+    * Start tracking version information for given table.
+    * This must be called before any other operations can be performed on this table.
+    */
+  def init(table: TableName): F[Unit]
+
+  /** Get details about partition versions in a table. */
+  def currentVersion(table: TableName): F[TableVersion]
+
+  /** Get a description of which version to write to next for the given partitions of a table. */
+  def nextVersions(table: TableName, partitions: List[Partition]): F[List[PartitionVersion]]
```

Comment:
Why does this return multiple versions?

Reply:
It's a version per partition. So if you're inserting a dataset that contains three days worth of data, then we need to get the appropriate version for each date partition that this will be written to.

Reply:
So the delta between contiguous versions is only ever going to be a single partition?

Reply:
Not sure if I understand the question, but a single update to a table can contain a number of updated partitions. Does that make sense?

Reply:
I think I've misread the cardinality:

> It's a version per partition.

I took that to mean that we get a new version for each updated partition.